src/vm/vm_core.rs

```diff
@@ -729,6 +729,19 @@ impl VirtualMachine {
             .write_arg(&mut self.memory, ptr, arg, Some(&self.prime))
     }
 
+    ///Gets `n_ret` return values from memory
+    pub fn get_return_values(
+        &self,
+        n_ret: usize,
+    ) -> Result<Vec<Option<Cow<MaybeRelocatable>>>, MemoryError> {
+        let addr = &self
+            .run_context
+            .get_ap()
+            .sub(n_ret)
+            .map_err(|_| MemoryError::NumOutOfBounds)?;
+        self.memory.get_range(&addr.into(), n_ret)
```

Comment:
```suggestion
    ) -> Result<Vec<Option<MaybeRelocatable>>, MemoryError> {
        let addr = &self
            .run_context
            .get_ap()
            .sub(n_ret)
            .map_err(|_| MemoryError::NumOutOfBounds)?;
        self.memory.get_range(&addr.into(), n_ret).iter()
            .map(|x| x.map(|val| val.to_owned()))
            .collect()
```

Reply:
The suggestion above is because it's better not to leak the `Cow` onto users.

Reply:
Ohh I see. Thanks @Oppen! :D